lib/core/src/Cardano/Wallet.hs

```diff
@@ -765,3 +768,25 @@ newWalletLayer tracer bp db nw tl = do
                         return $ checkPassphrase pwd hpwd
                     return xprv
         action xprv
+
+
+{---------------------------------------------------------------------------
+                                 Helpers
+  Defined outside 'WalletLayer' since they don't benefit from being "layered".
+---------------------------------------------------------------------------}
+
+-- | Regularly poll the wallet to monitor it's syncing progress. Block until the
+-- wallet reaches 100%.
+waitForWalletSync
+    :: WalletLayer s t
+    -> Trace IO Text
```

Comment:
Or should we put this inside the `WalletLayer`? We could use the `WalletLayer`'s tracer then.